File: Decorator/Java/src/condiments/Whip.java
package eu.svetlin.design.patterns.decorator.condiments;

import eu.svetlin.design.patterns.decorator.beverages.Beverage;

public class Whip extends CondimentDecorator {

    public Whip(Beverage beverage){
        this.beverage = beverage;
    }
    @Override
    public double cost() {
        return beverage.cost() + .10;
    }

    @Override
    public String getDescription() {
        return beverage.getDescription() + ", Whip";
    }
}

File: Decorator/TypeScript/src/condiments/Whip.ts
import Beverage from '../beverages/Beverage';
import CondimentDecorator from './CondimentDecorator';

class Whip extends CondimentDecorator {
  constructor(beverage: Beverage) {
    super();
    this.beverage = beverage;
  }
  getDescription() {
    return this.beverage.getDescription() + ', Whip';
  }

  cost(): number {
    return this.beverage.cost() + 0.1;
  }
}

export default Whip;
